dcc-portal-ui/app/scripts/oncogrid/js/controller.js

```diff
@@ -0,0 +1,118 @@
+
+        Settings.get().then(function(settings) {
+            _this.downloadEnabled = settings.downloadEnabled || false;
+        });
+
+        _this.downloadDonorData = function(id) {
+            $modal.open({
+                templateUrl: '/scripts/downloader/views/request.html',
+                controller: 'DownloadRequestController',
+                resolve: {
+                filters: function() { return {donor:{id:{is:[Extensions.ENTITY_PREFIX + id]}}}; }
+                }
+            });
+        };
+
+        _this.clusterData = function () {
+            $scope.grid.cluster();
+        };
+
+        // Export the subset(s), materialize the set along the way
+        $scope.exportSet = function (id) {
+            SetService.exportSet(id);
+        };
+
+        _this.reloadGrid = function () {
+            $scope.grid.destroy();
+            $scope.cleanActives();
+            $('#grid-button').addClass('active');
+            $scope.grid = new OncoGrid(_.cloneDeep($scope.params));
+            $scope.grid.render();
+            $scope.grid.removeDonors(function(d){return d.score === 0;});
+            $scope.grid.removeGenes(function(d){return d.score === 0;});
+                
+            if ($scope.crosshairMode) {
+                _this.crosshair();
+            }
+        };
+
+        _this.heatMap = function () {
+            $('#heat-button').toggleClass('active');
+            $('#og-variant-legend').toggle();
+            $('#og-heatmap-legend').toggle();
+
+            $scope.grid.toggleHeatmap();
+        };
+
+        _this.gridLines = function () {
+            $('#grid-button').toggleClass('active');
+            $scope.grid.toggleGridLines();
+        };
+
+        _this.crosshair = function () {
+            $scope.crosshairMode = $scope.crosshairMode ? false : true;
+            $('#og-crosshair-message').toggle();
+            $('#crosshair-button').toggleClass('active');
+            var gridDriv = $('#oncogrid-div');
+            gridDriv.toggleClass('og-pointer-mode'); gridDriv.toggleClass('og-crosshair-mode');
+            $scope.grid.toggleCrosshair();
+        };
+
+        function exitFullScreen() {
+            if (document.exitFullscreen) {
+                document.exitFullscreen();
+            } else if (document.mozCancelFullScreen) {
+                document.mozCancelFullScreen();
+            } else if (document.webkitExitFullscreen) {
+                document.webkitExitFullscreen();
+            }
+        }
+
+        function enterFullScreen() {
+            var element = document.getElementById('oncogrid-container');
+            if (element.requestFullscreen) {
+                element.requestFullscreen();
+            } else if (element.mozRequestFullScreen) {
+                element.mozRequestFullScreen();
+            } else if (element.webkitRequestFullScreen) {
+                element.webkitRequestFullScreen(Element.ALLOW_KEYBOARD_INPUT);
+            }
+        }
```

Comment:
Could we move this method into a utility file? Perhaps even just declare it as a global variable for now (until we can require in modules)

I could reuse this for the survival graph

Have the element be passed in

```js
function enterFullScreen(element) {
```